base/complex.jl

```diff
@@ -56,11 +56,24 @@ Return both the real and imaginary parts of the complex number `z`.
 """
 reim(z) = (real(z), imag(z))
 
-real{T<:Real}(::Type{T}) = T
-real{T<:Real}(::Type{Complex{T}}) = T
+"""
+    real(N::Type)
 
-complex{T<:Real}(::Type{T}) = Complex{T}
-complex{T<:Real}(::Type{Complex{T}}) = Complex{T}
+Gets the type that represents the real part of a value of type `N`.
+e.g: for `N <: Complex{T}`, returns `T`.
```

Comment:
Since `Compex{T}` is concrete, `<:` is superfluous.  Just use `N == Complex{T}`.